src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -96,9 +96,26 @@ EngineShoutcast::EngineShoutcast(ConfigObject<ConfigValue>* _config)
 }
 
 EngineShoutcast::~EngineShoutcast() {
+    int l_iTime = 0;
     m_pShoutcastEnabled->set(0);
     m_readSema.release();
-    wait(); // until the thread ends.
+
+    // Because Thread QThread msec doesn't blocking
+    // Wait 20 * 100 msec which is ~2 seconds for disconnection. If anything ain't happening
+    // it's not going to happen!
+    while (l_iTime < 20) {
```

Comment:
This burns some CPU cycles. Why we cant use `wait(2000)` here? 

If we assume that the thread does not die reliable once we have set m_pShoutcastEnabled->set(0);
We have also a problem when the user disconnects from shoutcast server. 
What happens if we call later start() again, but we have still locked thread around? 

This issue is probably more annoying than the deadlock at shout down, which is "fixed" here.